### PR TITLE
Merge new app version, #356 done.

### PR DESCRIPTION
**Pull request by DmitryMorozov228**

### CODE REVIEW

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Softeq.NetKit.Chat.Data.Persistent.Sql.csproj [hunk 1]
```diff
@@ -17,4 +17,8 @@
     <ProjectReference Include="..\Softeq.NetKit.Chat.Domain\Softeq.NetKit.Chat.Domain.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Remove="Repositories\DirectChannelsRepository.cs" />
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Softeq.NetKit.Chat.Data.Persistent.Sql.csproj]: Do we need to remove this file on compilation?

File: src/Softeq.NetKit.Chat.Domain.Services/DomainServices/ChannelService.cs [hunk 2]
```diff
@@ -289,33 +296,33 @@ public async Task<ChannelResponse> CloseChannelAsync(string saasUserId, Guid cha
...
-                var channelMember = await UnitOfWork.ChannelMemberRepository.GetChannelMemberAsync(creator.Id, channel.Id);
+                var channelMember = await UnitOfWork.ChannelMemberRepository.GetChannelMemberAsync(currentUser.Id, channel.Id);
                 if (channel.Type == ChannelType.Direct)
                 {
                     //TODO: Extend existing query to make one single request to get all data for both direct and group channel
-                    var members = await UnitOfWork.ChannelMemberRepository.GetChannelMembersAsync(channel.Id);
-                    var directChannelMember = members.First(x => x.MemberId != creator.Id);
-                    var member = await UnitOfWork.MemberRepository.GetMemberByIdAsync(directChannelMember.MemberId);
+                    var members = await UnitOfWork.ChannelMemberRepository.GetChannelMembersWithMemberDetailsAsync(channel.Id);
+                    var directChannelMember = members.First(x => x.MemberId != currentUser.Id);
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Domain.Services/DomainServices/ChannelService.cs]: Can we replace it with SingleOrDefault and throw an exception if it null?

File: src/Softeq.NetKit.Chat.Domain/DomainModels/Member.cs [hunk 3]
```diff
@@ -19,6 +19,7 @@ public class Member : IBaseEntity<Guid>
         public bool IsAfk { get; set; }
 
         public bool IsActive { get; set; }
+        public bool IsDeleted { get; set; }
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Domain/DomainModels/Member.cs]: Not sure if it's more proper way for doing this, but let's consider using nullable "DeletedAt" instead. This way we will be able to know if the member is deleted + deletion datetime.

File: src/Softeq.NetKit.Chat.Domain/DomainModels/Message.cs [hunk 4]
```diff
@@ -18,6 +18,7 @@ public class Message : IBaseEntity<Guid>, ICreated
         public DateTimeOffset? Updated { get; set; }
         public MessageType Type { get; set; }
         public AccessibilityStatus AccessibilityStatus { get; set; }
+        public bool IsDeleted { get; set; }
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Domain/DomainModels/Message.cs]: The same as for Member's IsDeleted.

File: src/Softeq.NetKit.Chat.Domain/TransportModels/Response/Member/MemberSummary.cs [hunk 5]
```diff
@@ -21,8 +21,9 @@ public class MemberSummary
 
         public bool IsAfk { get; set; }
         public bool IsActive { get; set; }
+        public bool IsDeleted { get; set; }
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Domain/TransportModels/Response/Member/MemberSummary.cs]: The same as for Member's IsDeleted.

File: src/Softeq.NetKit.Chat.Web/TransportModels/Request/Channel/CreateDirectChannelRequest.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace FQ.DEEMA.Chat.Web.TransportModels.Request.Channel
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Web/TransportModels/Request/Channel/CreateDirectChannelRequest.cs]: Why are we using this namespace part? - "FQ.DEEMA."
Let's find all such namespaces and correct it.

File: src/Softeq.NetKit.Chat.Web/TransportModels/Validators/Channel/CreateChannelRequestValidator.cs [hunk 7]
```diff
@@ -11,6 +11,7 @@ public class CreateChannelRequestValidator : AbstractValidator<CreateChannelRequ
         public CreateChannelRequestValidator()
         {
             RuleFor(x => x.Name).NotNull().NotEmpty();
+            RuleFor(x => x.Name).Must(name => name != null && !name.Contains(' ')).WithMessage(x => $"{nameof(x.Name)} cannot contain spaces.");
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Web/TransportModels/Validators/Channel/CreateChannelRequestValidator.cs]: Why not use string.IsNullOrWhiteSpace or .NotNull().NotEmpty()?

File: src/Softeq.NetKit.Chat.Domain.Services/DomainServices/ChannelService.cs [hunk 8]
```diff
@@ -271,12 +283,7 @@ public async Task<ChannelResponse> CloseChannelAsync(string saasUserId, Guid cha
                 throw new NetKitChatNotFoundException($"Unable to close channel. Member {nameof(saasUserId)}:{saasUserId} is not found.");
             }
 
-            if (channel.Type == ChannelType.Direct)
-            {
-                throw new NetKitChatInvalidOperationException($"Unable to close direct channel {nameof(channelId)}:{channelId}.");
-            }
-            
-            if (member.Id != channel.CreatorId)
+            if (member.Id != channel.CreatorId && channel.Type != ChannelType.Direct)
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Domain.Services/DomainServices/ChannelService.cs]: Do we need to check "member" and "channel" for null?

File: src/Softeq.NetKit.Chat.Domain.Services/DomainServices/ChannelService.cs [hunk 9]
```diff
@@ -168,15 +177,18 @@ public async Task<ChannelSummaryResponse> CreateDirectChannelAsync(CreateDirectC
             using (var transactionScope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
             {
                 await UnitOfWork.ChannelRepository.AddChannelAsync(newChannel);
+
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Domain.Services/DomainServices/ChannelService.cs]: Did you merge latest develop into your branch? We got rid of this lines "using (var transactionScope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))" in develop and extracted it in the UnitOfWork.

File: src/Softeq.NetKit.Chat.Tests/Integration/RepositoryTests/MemberRepositoryTests.cs [hunk 10]
```diff
@@ -369,5 +370,60 @@ public async Task GetAllMembersByChannelIdAsync_ShouldReturnAllMembersInChannel(
             // Assert
             membersFromChannel.Should().BeEquivalentTo(expectedMembers);
         }
+
+        [Fact]
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Tests/Integration/RepositoryTests/MemberRepositoryTests.cs]: Did you run unit tests? Is everything works as expected? We need to get latest develop and rerun the unit tests.

File: src/Softeq.NetKit.Chat.Domain.Services/DomainServices/ChannelService.cs (hunk 2 above)
Comment (DmitryMorozov228) [reply src/Softeq.NetKit.Chat.Domain.Services/DomainServices/ChannelService.cs]: done

File: src/Softeq.NetKit.Chat.Web/TransportModels/Request/Channel/CreateDirectChannelRequest.cs (hunk 6 above)
Comment (DmitryMorozov228) [reply src/Softeq.NetKit.Chat.Web/TransportModels/Request/Channel/CreateDirectChannelRequest.cs]: done

File: src/Softeq.NetKit.Chat.Web/TransportModels/Validators/Channel/CreateChannelRequestValidator.cs [hunk 11]
```diff
@@ -11,7 +11,6 @@ public class CreateChannelRequestValidator : AbstractValidator<CreateChannelRequ
         public CreateChannelRequestValidator()
         {
             RuleFor(x => x.Name).NotNull().NotEmpty();
-            RuleFor(x => x.Name).Must(name => name != null && !name.Contains(' ')).WithMessage(x => $"{nameof(x.Name)} cannot contain spaces.");
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Web/TransportModels/Validators/Channel/CreateChannelRequestValidator.cs]: Why did you remove the message? Do we need it?

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelMemberRepository.cs [hunk 12]
```diff
@@ -108,10 +108,18 @@ public async Task<IReadOnlyCollection<ChannelMember>> GetChannelMembersWithMembe
...
-                                 FROM ChannelMembers cm
-                                 INNER JOIN Members m ON m.Id = cm.MemberId
-                                 WHERE ChannelId = @channelId";
+                var sqlQuery = $@"
+                    SELECT 
+                        {nameof(ChannelMember.ChannelId)}, 
+                        {nameof(ChannelMember.MemberId)}, 
+                        {nameof(ChannelMember.LastReadMessageId)}, 
+                        {nameof(ChannelMember.IsMuted)}, 
+                        {nameof(ChannelMember.IsPinned)}
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelMemberRepository.cs]: We need to add Members to SELECT as we are using INNER JOIN here.

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelRepository.cs [hunk 13]
```diff
@@ -341,12 +387,20 @@ public async Task<Guid> GetDirectChannelForMembersAsync(Guid member1Id, Guid mem
...
-                                    HAVING COUNT(cm.MemberId) = 2";
+                var sqlQuery = $@"
+                    SELECT 
+                        c.{nameof(Channel.Id)}
+                    FROM 
+                        Channels c
+                    INNER JOIN ChannelMembers cm 
+                        ON c.{nameof(Channel.Id)} = cm.{nameof(ChannelMember.ChannelId)}
+                    WHERE
+                        (cm.{nameof(ChannelMember.MemberId)} = @{nameof(member1Id)} OR cm.{nameof(ChannelMember.MemberId)} = @{nameof(member2Id)})
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelRepository.cs]: I do not see the second part of the initial request: ((cm.MemberId = @member1Id OR cm.MemberId = @member2Id) **AND (c.CreatorId = @member1Id OR c.CreatorId = @member2Id))**

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/MemberRepository.cs [hunk 14]
```diff
@@ -204,10 +278,25 @@ public async Task<IReadOnlyCollection<Member>> GetAllMembersByChannelIdAsync(Gui
...
+                        m.{nameof(Member.Email)}, 
+                        m.{nameof(Member.IsAfk)}, 
+                        m.{nameof(Member.IsBanned)}, 
+                        m.{nameof(Member.LastActivity)}, 
+                        m.{nameof(Member.LastNudged)}, 
+                        m.{nameof(Member.Name)}, 
+                        m.{nameof(Member.PhotoName)}, 
+                        m.{nameof(Member.Role)}, 
+                        m.{nameof(Member.SaasUserId)}, 
+                        m.{nameof(Member.Status)}
```
Comment (mr-archie) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/MemberRepository.cs]: We need to add ChannelMembers into SELECT statement as we are using INNER JOIN here.

Comment (DmitryMorozov228) [reply src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/MemberRepository.cs]: No, we didn't

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelMemberRepository.cs (hunk 12 above)
Comment (DmitryMorozov228) [reply src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelMemberRepository.cs]: done

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelRepository.cs (hunk 13 above)
Comment (DmitryMorozov228) [reply src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelRepository.cs]: done
